packages/loopring_v3/circuit/Circuits/WithdrawCircuit.h

```diff
@@ -362,9 +353,8 @@ class WithdrawCircuit : public BaseTransactionCircuit {
 
   const VariableArrayT getPublicData() const {
     return flattenReverse({type.bits, owner.bits, accountID.bits, tokenID.bits,
-                           feeTokenID.bits, amount.bits, fFee.bits(), to.bits,
-                           dataHash.bits, minGas.bits, validUntil.bits,
-                           nonce.bits});
+                           amount.bits, feeTokenID.bits, fFee.bits(),
+                           dataHash.bits, nonce.bits});
   }
```

Comment:
Missing validUntil.bits？

Reply:
Removed to save a couple of bits for optimization.

If `validUntil` would be put here, then yes, we don't have to check `validUntil` onchain. This is how it worked before this PR, but I changed it to save on some sha256 hashing constraints.

I removed it because it's technically not data-availability data, it's not never needed to reconstruct the Merkle tree.

It this is too confusing I can add it back, but it does save quite some constraints (~2,000 constraints/tx).